src/activities/generate-version-json.ts

```diff
@@ -56,6 +33,7 @@ export const generateVersionJson: GenerateVersionJson = async ({ importContentRe
     .then((obj) => obj.Body?.transformToString() ?? '');
 
   const articleStruct = parseJsonContentToProcessedArticle(json);
+  const msid = `${version.doi.split('/').pop()}v${version.versionIdentifier}`;
```

Comment:
These are some very elife assumptions - the idea that the docmap parser returns the "manuscript id" feels like an important one to keep than work around. If you were seeing problems with the id, we should work to fix that in the parsing step.

Reply:
The issue is import-docmap was passing msid as the id of the docmap, so either way it is 'wrong'

Reply:
But the msid of a manuscript should be the id of the docmap. As in - the ID parsed from a docmap should be about 1 manuscript, multiple versions, which is what an msid refers to eLife.

Maybe this is where https://github.com/elifesciences/enhanced-preprints-issues/issues/621 could help?

So taking 85111 as an example (as a real multi-version RPP):

> ```
> msid: 85111
> doi: 10.7554/85111
> versionId(entifier): 2
> versionDoi: 10.7554/85111.2
> routeId: 85111v2
> contentId: 85111/v2
> ```

A (slightly fixed) version of `docmap-ts` parsing output for the latest example from Hazal:
```
> yarn run -s ts-node src/parse.ts '/Users/scottaubrey/Downloads/sample_docmap_for_85111 (1).json' | jq '.versions = ([.versions[] | del(.peerReview)])'
{
  "id": "85111",
  "versions": [
    {
      "doi": "10.7554/eLife.85111.1",
      "id": "85111",
      "versionIdentifier": "1",
      "preprint": {
        "id": "10.1101/2022.11.08.515698",
        "doi": "10.1101/2022.11.08.515698",
        "versionIdentifier": "2"
      },
      "sentForReviewDate": "2022-11-28T11:30:05.000Z",
      "reviewedDate": "2023-01-23T14:34:45.299Z",
      "status": "Enhanced Preprint"
    },
    {
      "doi": "10.7554/eLife.85111.2",
      "id": "85111",
      "versionIdentifier": "2",
      "preprint": {
        "id": "10.1101/2022.11.08.515698",
        "doi": "10.1101/2022.11.08.515698",
        "versionIdentifier": "3"
      },
      "sentForReviewDate": "2023-03-23T10:52:45.000Z",
      "reviewedDate": "2023-04-14T13:42:25.781Z",
      "status": "Enhanced Preprint"
    }
  ]
}
```

So all these values should be straightforward translations to the EPP server API:

```
    msid: docmap.id,
    doi: docmap[version].doi,
    versionIdentifier: docmap[version].versionIdentifier,
    versionDoi: docmap[version].doi,
    preprintDoi: docmap[version].preprint.doi,
    preprintUrl: docmap[version].preprint.url,
    preprintPosted: docmap[version].preprint.publishedDate,
    sentForReview: docmap[version].sentForReviewDate,
    published: versiondocmap[version].publishedDate,
```

As per the conversation Friday, we should store `id`/`_id` for server with an format like routeId or contentId, or as a composite primary key of `msid` and `versionIdentifier`. If using a string - this would need to be constructed right now I guess.

(as a side note: I'm actually starting to think I misunderstood `versionIdentifier`, and it should be a globally unique ID for the version, not a id-dependant version identifier (i.e. 12345v1, not 1). That would be a breaking change though, but would really help us here.)

Hope that all makes sense, happy to chat as always.